### PR TITLE
[PyTorch][C10] Implemented migration rules for 3 PyTorch c10::cuda::* namespace APIs

### PR DESCRIPTION
**Pull request by TejaX-Alaghari**
Introducing migration rule for 3 PyTorch c10::cuda::* namespace APIs

### CODE REVIEW

File: clang/lib/DPCT/APINames_PyTorch.inc [hunk 1]
```diff
@@ -0,0 +1,39 @@
+//===------------------------ APINames_PyTorch.inc ------------------------===//
```
Comment (zhiweij1) [inline clang/lib/DPCT/APINames_PyTorch.inc]: I think there is no need to introduce this file. pytorch framework API is not cuda API.

File: clang/test/dpct/pytorch_c10_cuda_ns.cu [hunk 2]
```diff
@@ -0,0 +1,43 @@
+// RUN: cat %s > %T/pytorch_c10_cuda_ns.cu
+// RUN: cat %S/user_defined_rule_pytorch.yaml > %T/user_defined_rule_pytorch.yaml
+// RUN: cd %T
+// RUN: rm -rf %T/pytorch_c10_cuda_ns_output
+// RUN: mkdir %T/pytorch_c10_cuda_ns_output
+// RUN: dpct -out-root %T/pytorch_c10_cuda_ns_output pytorch_c10_cuda_ns.cu --cuda-include-path="%cuda-path/include" --usm-level=none --rule-file=user_defined_rule_pytorch.yaml  -- -x cuda --cuda-host-only
+// RUN: FileCheck --input-file %T/pytorch_c10_cuda_ns_output/pytorch_c10_cuda_ns.dp.cpp --match-full-lines pytorch_c10_cuda_ns.cu
+// RUN: %if build_lit %{icpx -c -fsycl -DBUILD_TEST  %T/pytorch_c10_cuda_ns_output/pytorch_c10_cuda_ns.dp.cpp -o %T/pytorch_c10_cuda_ns_output/pytorch_c10_cuda_ns.dp.o %}
```
Comment (zhiweij1) [inline clang/test/dpct/pytorch_c10_cuda_ns.cu]:
```suggestion
// RUN: %if build_lit %{icpx -c -fsycl -DNO_BUILD_TEST  %T/pytorch_c10_cuda_ns_output/pytorch_c10_cuda_ns.dp.cpp -o %T/pytorch_c10_cuda_ns_output/pytorch_c10_cuda_ns.dp.o %}
```

File: clang/test/dpct/pytorch_c10_cuda_ns.cu [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+#ifndef BUILD_TEST
```
Comment (zhiweij1) [inline clang/test/dpct/pytorch_c10_cuda_ns.cu]:
```suggestion
#ifndef NO_BUILD_TEST
```

File: clang/lib/DPCT/APINames_PyTorch.inc (hunk 1 above)
Comment (TejaX-Alaghari) [reply clang/lib/DPCT/APINames_PyTorch.inc]: Okay, I'll remove this file.

**Review by zhimingwang36 — APPROVED** [review]

**Review by zhiweij1 — APPROVED** [review]

**Pull request merged**